include/geom/elem.h

```diff
@@ -1573,6 +1609,45 @@ class Elem : public ReferenceCountedObject<Elem>,
 
 
 // ------------------------------------------------------------
+// Elem helper classes
+
+#ifdef LIBMESH_ENABLE_AMR
+class
+Elem::ChildRefIter : public PPIter<Elem>
+{
+public:
+  ChildRefIter (Elem * const * childpp) : PPIter<Elem>(childpp) {}
+};
+
+
+class
+Elem::ConstChildRefIter : public PPIter<const Elem>
+{
+public:
+  ConstChildRefIter (const Elem * const * childpp) : PPIter<const Elem>(childpp) {}
+};
+
+
+inline
+SimpleRange<Elem::ChildRefIter> Elem::child_ref_range()
+{
+  libmesh_assert(_children);
+  return {_children, _children + this->n_children()};
```

Comment:
I forget what construction syntax is called, maybe "uniform initilization"? 

I don't think we explicitly test for this in *any* of the configure tests, although I sure hope we don't come across a compiler that doesn't support this yet supports everything else we do test for....

Reply:
Uniform initialization.  I didn't want to put any explicit intermediaries here because I was wary that doing so incorrectly might force an actual temporary construction, make return value elision harder, or in some other way pessimize the code.

You may be right to be concerned, though?  [It looks like](http://en.cppreference.com/w/cpp/compiler_support) initializer lists were implemented *after* ranged for loops by some compilers; in particular users of clang 3.0, MSVC++ 17.0, and possibly Intel 13.0 could be bitten at compile time rather than configure time.